packages/lesswrong/components/common/RecommendedReading.jsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import { Components, registerComponent } from 'meteor/vulcan:core';
+import { Link } from 'react-router';
+import withUser from '../common/withUser';
+import { withStyles } from '@material-ui/core/styles';
+import Hidden from '@material-ui/core/Hidden';
+
+const styles = theme => ({
+  smallScreenRecommendedReading: {
+    maxWidth: 720,
+    margin: "0 auto",
+  },
+});
+
+
+const RecommendedReading = ({currentUser, classes}) => {
+  const sectionTitle = currentUser ? "Recommended Sequences" : "Recommended Reading";
+  
+  const sectionWrapper = (recommendedContent) => (
+    <Components.Section
+      title={sectionTitle}
+      titleLink="/library"
+      titleComponent= {<Components.SectionSubtitle to="/library">
+        <Link to="/library">Sequence Library</Link>
+      </Components.SectionSubtitle>}
+    >{recommendedContent}</Components.Section>
+  );
+  
+  const omittedSectionWrapper = (content) => (
+    <div className={classes.smallScreenRecommendedReading}>
+      {content}
+    </div>
+  );
+  
+  if (currentUser) {
+    const suggestedSequences = count => (
+      <Components.SequencesGridWrapper
+        terms={{view:"curatedSequences", limit: count}}
+        showAuthor={true}
+        showLoadMore={false}
+        className={classes.frontpageSequencesGridList}
+      />
+    );
+    
+    return (<React.Fragment>
+      <Hidden smDown implementation="css">
+        {sectionWrapper(
```

Comment:
This SectionWrapper seems maybe better to factor out than the other one, but I think you should not implement it as a HoC, but just as a component with children. I.e. make the code the following: 

```
const SectionWrapper = ({children}) => {
  <Components.Section ... >
      {children}
</Components.Section>
}
```
This makes the whole thing a lot easier to read and is more performant. 

When you do that, I would also create the SectionWrapper component outside of the render function, since you don't need the RecommendedReading scope here at all, and that saves a bunch of function declarations and will probably save on rerendering on props changes.

Reply:
I guess using React's syntactic sugar here doesn't hurt. I don't think it's right to move this to top-level, though, both because it'd be violating the usual one-component-per-file rule, and also because it falsely suggests that the component is general enough to be used elsewhere.

Reply:
Huh, I don't care much about the one component per file rule, and generally prefer things to be at the top level, since it's then easy to move in case you do actually want to make it it's own file. But not moving it there seems fine.